platform-server-kernel/src/main/java/uno/cod/platform/server/core/service/SubmissionService.java

```diff
@@ -119,8 +122,41 @@ public void run(User user, UUID taskId, MultipartFile file, String language) thr
         appClientConnection.send(user.getId(), runtimeClient.postToRuntime(task.getRunner().getName(), form).toString());
     }
 
-    public boolean testOutput(UUID testId, MultipartFile file) throws IOException {
-        Test test = testRepository.findOneWithRunner(testId);
+    public List<OutputTestResultDto> testOutput(UUID resultId, UUID taskId, MultipartFile[] files) throws IOException {
+        Result result = resultRepository.findOne(resultId);
+        if (result == null) {
+            throw new IllegalArgumentException("result.invalid");
+        }
+
+        Task task = taskRepository.findOneWithTests(taskId);
+        if (task == null) {
+            throw new IllegalArgumentException("task.invalid");
+        }
+
+        TaskResult taskResult = taskResultService.findByTaskAndResult(taskId, resultId);
+        Submission submission = new Submission();
+        submission.setTaskResult(taskResult);
+        submission.setSubmissionTime(ZonedDateTime.now());
+        submission = repository.save(submission);
+
+        List<OutputTestResultDto> testResults = new ArrayList<>();
```

Comment:
We know the capacity needed for this `ArrayList` beforehand (it is `files.length`), so please pass it when constructing the object.